moxin-protocol/src/protocol.rs

```diff
@@ -1,11 +1,12 @@
-use std::sync::mpsc::Sender;
-use anyhow::Result;
 use crate::data::*;
 use crate::open_ai::*;
+use anyhow::Result;
+use std::sync::mpsc::Sender;
+use std::sync::Arc;
 
 #[derive(Clone, Debug)]
 pub enum FileDownloadResponse {
-    Progress(FileID, f32),
+    Progress(Arc<FileID>, f32),
```

Comment:
See my related comment with an idea to avoid adding the `Arc` here, so the protocol is free of shared state stuff like Arc, Mutex, etc

Reply:
Okay, I'll revert it in my next commit. Maybe there's no need to consider these issues for now.

Reply:
I have reverted it.